airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -277,11 +281,19 @@ def execute(self, context):
                     pod,
                     startup_timeout=self.startup_timeout_seconds,
                     get_logs=self.get_logs)
+            except AirflowException:
+                if self.log_events_on_failure:
+                    for event in launcher.read_pod_events(pod).items:
+                        self.log.error("Pod Event: %s - %s", event.reason, event.message)
+                raise
             finally:
                 if self.is_delete_operator_pod:
                     launcher.delete_pod(pod)
 
             if final_state != State.SUCCESS:
+                if self.log_events_on_failure:
```

Comment:
I think we should protect against writing the logs twice. In case of Airflow Exception thrown it is more than likely that the final_state will not be a success, so the logs will be written twice. I think we should have a flag "logs_dumped" or smth like that an only write logs once.

Reply:
If we hit an exception at lines 285-288, we would log the pod events, then re-raise the exception which shouldn't allow execution to the logging at lines 294-296 - so I don't think it's possible to log them twice - unless I'm missing something here, I haven't had coffee yet :)

Reply:
You are right. No worries. I am after my 3rd coffee and still missed the `raise`